From 4c16dc70ca59ec0ae903748e756c2701b1673479 Mon Sep 17 00:00:00 2001
From: Adrian Perez de Castro <aperez@igalia.com>
Date: Wed, 2 Apr 2025 06:18:23 -0700
Subject: [PATCH 3/7] Introduce OS(ANDROID) to simplify preprocessor guards
 https://bugs.webkit.org/show_bug.cgi?id=290913

Reviewed by Carlos Garcia Campos.

Introduce a new WTF_OS_ANDROID macro, and use OS(ANDROID) guards
to replace (and simplify) the existing ones.

* Source/WTF/wtf/PlatformOS.h:
* Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp:
(WebCore::PlatformDisplay::angleEGLDisplay const):
* Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp:
(WebCore::FontCache::fontManager const):

Canonical link: https://commits.webkit.org/293096@main
---
 Source/WTF/wtf/PlatformOS.h                                | 5 +++++
 .../platform/graphics/angle/PlatformDisplayANGLE.cpp       | 2 +-
 Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp    | 7 +++++--
 3 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/Source/WTF/wtf/PlatformOS.h b/Source/WTF/wtf/PlatformOS.h
index a1cc02412378..406785cf8b4a 100644
--- a/Source/WTF/wtf/PlatformOS.h
+++ b/Source/WTF/wtf/PlatformOS.h
@@ -52,6 +52,11 @@
 #define WTF_OS_AIX 1
 #endif
 
+/* OS(ANDROID) - Android */
+#if defined(__ANDROID__) || defined(ANDROID)
+#define WTF_OS_ANDROID 1
+#endif
+
 /* OS(DARWIN) - Any Darwin-based OS, including macOS, iOS, iPadOS, macCatalyst, tvOS, watchOS and visionOS */
 #if defined(__APPLE__)
 #define WTF_OS_DARWIN 1
diff --git a/Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp b/Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp
index a6959bf5afd0..10643eac40e2 100644
--- a/Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp
+++ b/Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp
@@ -44,7 +44,7 @@ EGLDisplay PlatformDisplay::angleEGLDisplay() const
 
     Vector<EGLint> displayAttributes {
         EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_OPENGLES_ANGLE,
-#if !defined(__ANDROID__) && !defined(ANDROID)
+#if !OS(ANDROID)
         EGL_PLATFORM_ANGLE_DEVICE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_DEVICE_TYPE_EGL_ANGLE,
         EGL_PLATFORM_ANGLE_NATIVE_PLATFORM_TYPE_ANGLE, m_anglePlatform.value(),
 #endif
diff --git a/Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp b/Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp
index 92a88f6b5691..48410a334ff7 100644
--- a/Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp
+++ b/Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp
@@ -29,8 +29,11 @@
 #include "Font.h"
 #include "FontDescription.h"
 #include "StyleFontSizeFunctions.h"
-#if defined(__ANDROID__) || defined(ANDROID)
+#if OS(ANDROID)
 #include <skia/ports/SkFontMgr_android.h>
+#elif PLATFORM(WIN)
+#include <dwrite.h>
+#include <skia/ports/SkTypeface_win.h>
 #else
 #include <skia/ports/SkFontMgr_fontconfig.h>
 #endif
@@ -52,7 +55,7 @@ void FontCache::platformInit()
 SkFontMgr& FontCache::fontManager() const
 {
     if (!m_fontManager) {
-#if defined(__ANDROID__) || defined(ANDROID)
+#if OS(ANDROID)
         m_fontManager = SkFontMgr_New_Android(nullptr);
 #else
         m_fontManager = SkFontMgr_New_FontConfig(FcConfigReference(nullptr));
-- 
2.43.0

